### PR TITLE
Vector of shared ptrs

### PR DESCRIPTION
**Pull request by Rajzell**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @Rajzell

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+}
+
+void sub10(int * pnt)
+{
+    if(pnt != nullptr)
+        *pnt =  *pnt - 10;
+}
+
+void sub10(std::vector<std::shared_ptr<int>> vec)
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:33]: do not copy, use reference!

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+std::vector<std::shared_ptr<int>> generate(int count)
+{
+    std::vector<std::shared_ptr<int>> vec(count); //initialize a vector of shared_pointers of size = count
+    for (int i=0; i<count; ++i){
+        vec[i] = (std::make_shared<int>(i));
+    }
+    return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec)
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:12]: do not copy, use reference!

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+{
+    for(auto element:vec){
+        std::cout << *element;
+    }
+}
+
+void add10(std::vector<std::shared_ptr<int>> vec)
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:19]: do not copy, use reference!

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**